qiskit_metal/analyses/quantization/lom_core_analysis.py

```diff
@@ -0,0 +1,1267 @@
+
+        system_type (str): name of the quantum subsystem
+        default_opts (dict): default options relevant to the type of quantum
+            subsystem. Provide any default values for the paremeters that
+            make_quantum() needs. In the case of qubit subsystems, make_quantum()
+            takes advantage of the scQbuits package (https://scqubits.readthedocs.io).
+            Consequently, the default options correspond to the default options of
+            the qubit classes in scQubits
+    """
+
+    system_type = 'FLUXONIUM'
+    default_opts = {'cutoff': 110, 'truncated_dim': 10}
+
+    # pylint: disable=protected-access
+    # pylint: disable=no-member
+    @set_builder_options
+    def make_quantum(self, subsystem: Subsystem):
+        cg = self.cg
+        c_inv_k = cg.C_inv_k
+
+        subsystem_idx = _find_subsystem_mat_index(cg,
+                                                  subsystem,
+                                                  single_node=True)
+        ss_idx = subsystem_idx[0]
+        subsystem.system_params['subsystem_idx'] = ss_idx
+
+        # EJ and EC are in MHz
+        EC = Convert.Ec_from_Cs(1 / c_inv_k[ss_idx, ss_idx])
+        Q_zpf = 2 * ele
+        builder_options = self.builder_options
+        builder_options.EC = EC
+        fluxonium = scq.Fluxonium(**builder_options.view_as(scq.Fluxonium))
+
+        subsystem._quantum_system = fluxonium
+        subsystem._h_params = dict(EJ=builder_options.EJ,
+                                   EC=builder_options.EC,
+                                   EL=builder_options.EL,
+                                   flux=builder_options.flux,
+                                   Q_zpf=Q_zpf)
+        subsystem._h_params['default_charge_op'] = Operator(
+            fluxonium.n_operator(), False)
+
+
+class TLResonatorBuilder(QuantumBuilder):
+    """
+    Concrete builder class for a type of quantum subsystem
+    Each subsystem of a composite system calls make_quantum() method of this class
+    to map LOM analysis results, the reduced L and C matrix elements to hamiltonian
+    parameters, such as Ej, Ec and etc
+
+    To create your own custom quantum builder, follow the patterns of the existing
+    ones, such this one, or the other ones implemented in this library
+
+        system_type (str): name of the quantum subsystem
+        default_opts (dict): default options relevant to the type of quantum
+            subsystem. Provide any default values for the paremeters that
+            make_quantum() needs. In the case of qubit subsystems, make_quantum()
+            takes advantage of the scQbuits package (https://scqubits.readthedocs.io).
+            Consequently, the default options correspond to the default options of
+            the qubit classes in scQubits
+    """
+
+    system_type = 'TL_RESONATOR'
+    default_opts = {
+        'Z0': 50,
+        'vp': 'use_design',  # phase velocity
+        'design': {
```

Comment:
We may later hav ea things that pull sthese aprams from the params of a QCompoentn